homeassistant/components/camera/__init__.py

```diff
@@ -530,6 +530,15 @@ def supported_features(self) -> CameraEntityFeature:
         """Flag supported features."""
         return self._attr_supported_features
 
+    @property
+    def supported_features_compat(self) -> CameraEntityFeature:
+        """Return the supported features as CameraEntityFeature.
+
+        Remove this compatibility shim in 2025.1 or later.
+        """
+        features = self.supported_features
+        return CameraEntityFeature(features) if type(features) is int else features  # noqa: E721
```

Comment:
Could we add a warning log entry if the type was int like we do it in the deprecation helper? The best would probably to reuse the functions defined there.
https://github.com/home-assistant/core/blame/dev/homeassistant/helpers/deprecation.py

Reply:
I'm not sure any of those fit this case but we could make a new helper

Reply:
Probably will put it in the base entity class for these cases as it makes it easier to track if we already warned and we already have that pattern.

We likely need this for about 26 platforms but it only matters for the 9 that use the faster checks in terms of 2024.1.x so I'll focus on those and we can add it for the other ones later.

Reply:
Looks like I'm going to need to add tests for each platform as well since all the tests use the new enums and we don't have anything using the deprecated values

Reply:
Once I have a pattern at least it will be the nearly impossible he same PR 9 times so hopefully not to much review work

Reply:
It turns out the int values were being used directly to avoid importing the constants for fear they may disappear so the constant import will not generate a deprecation warning either. 🙃

Reply:
I think a helper is the best :)
I would extend the function here and call the helper if we got an int value, which then creates the warning log entry.

Reply:
https://github.com/home-assistant/core/pull/106602